ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/subviews/ToolVolumesFromListEditor.java

```diff
@@ -0,0 +1,221 @@
+                String url = urlTextField.getCurrentValue();
+                Boolean readOnly = readOnlyCheckBox.getValue();
+                ToolVolumesFrom volumesFrom = listStore.get(0);
+                volumesFrom.setName(name);
+                volumesFrom.setNamePrefix(namePrefix);
+                volumesFrom.setTag(tag);
+                volumesFrom.setUrl(url);
+                volumesFrom.setReadOnly(readOnly);
+                listStore.update(volumesFrom);
+            }
+        });
+    }
+
+    @Override
+    public Editor<List<ToolVolumesFrom>> asEditor() {
+        return new ListStoreEditor<>(listStore);
+    }
+
+    public void addVolumesFrom() {
+        ToolAutoBeanFactory factory = GWT.create(ToolAutoBeanFactory.class);
+        ToolVolumesFrom volumesFrom = factory.getVolumesFrom().as();
+        getVolumesFromTag(volumesFrom);
+
+        volumesFrom.setName("");
+        volumesFrom.setUrl("");
+        volumesFrom.setTag("");
+        volumesFrom.setNamePrefix("");
+        volumesFrom.setReadOnly(true);
+
+        editing.cancelEditing();
+        listStore.add(0, volumesFrom);
+        int row = listStore.indexOf(volumesFrom);
+        editing.startEditing(new Grid.GridCell(row, 0));
+    }
+
+    private String getVolumesFromTag(ToolVolumesFrom volumesFrom){
+        if (volumesFrom != null){
+            final AutoBean<ToolVolumesFrom> volumesFromAutoBean = AutoBeanUtils.getAutoBean(volumesFrom);
+            String currentTag = volumesFromAutoBean.getTag(TOOL_VOLUMES_FROM_MODEL_KEY);
+            if (currentTag == null){
+                volumesFromAutoBean.setTag(TOOL_VOLUMES_FROM_MODEL_KEY,
+                                      String.valueOf(unique_volumes_from_id++));
+            }
+            return volumesFromAutoBean.getTag(TOOL_VOLUMES_FROM_MODEL_KEY);
+        }
+        return "";
+    }
+
+    public void deleteVolumesFrom() {
+        ToolVolumesFrom deleteVolumesFrom = grid.getSelectionModel().getSelectedItem();
+        if (deleteVolumesFrom != null) {
+            listStore.remove(listStore.findModelWithKey(getVolumesFromTag(deleteVolumesFrom)));
+        }
+    }
+
+    public List<ToolVolumesFrom> getVolumesFromList() {
+        List<ToolVolumesFrom> volumesFromList = Lists.newArrayList();
+        removeDisallowedId(volumesFromList);
+        return volumesFromList;
+    }
+
+    private void removeDisallowedId(List<ToolVolumesFrom> volumesFromList) {
+        for (ToolVolumesFrom volumeFrom : listStore.getAll()) {
+            volumeFrom.setId(null);
+            volumesFromList.add(volumeFrom);
+        }
+    }
+
+    public boolean isValid(){
+        boolean valid = true;
+        for (ToolVolumesFrom volumesFrom : listStore.getAll()){
+            if (Strings.isNullOrEmpty(volumesFrom.getName()) || Strings.isNullOrEmpty(volumesFrom.getNamePrefix())){
+                return false;
+            }
+        }
+        return valid;
```

Comment:
`valid` is always true. Either inline the value, or set `valid=false` in the if block above.